crates/uplc/src/parser.rs

```diff
@@ -24,6 +24,19 @@ pub fn program(src: &str) -> Result<Program<Name>, ParseError<LineCol>> {
     Ok(program)
 }
 
+pub fn term(src: &str) -> Result<Term<Name>, ParseError<LineCol>> {
+    // initialize the string interner to get unique name
+    let mut interner = Interner::new();
+
+    // run the generated parser
+    let mut term = uplc::term(src)?;
+
+    // assign proper unique ids in place
+    interner.term(&mut term);
```

Comment:
not really sure if we want to do this. but it should be fine, most inputs would likely just end up being constants anyways.

@nielstron @MicroProofs 
actually, should I only allow constants?

Reply:
I don't really see a benefit in restricting it to constants 🤷‍♂️